### PR TITLE
fix(core): Add algo seed encoding

### PR DESCRIPTION
**Pull request by CapnMigraine**
Add logic to convert a stellar seed to algo encoding

CLOSES TICKET: STLX-6599

### CODE REVIEW

File: modules/account-lib/src/coin/algo/seedEncoding.ts [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+    // Check if the checksum and the seed are equal
+    if (checksum.length !== decoded.checksum.length) {
+      return false;
+    }
+    return checksum.every((val, i) => val === decoded.checksum[i]);
+  }
+
+  /**
+   * @param seed
```
Comment (rliberal-simtlix) [inline modules/account-lib/src/coin/algo/seedEncoding.ts:46]: Nit: add method title or comment

File: modules/core/src/v2/coins/algo.ts [hunk 2]
```diff
@@ -348,6 +350,23 @@ export class Algo extends BaseCoin {
     return type === 'axfer';
   }
 
+  isStellarSeed(seed: string): boolean {
```
Comment (rliberal-simtlix) [inline modules/core/src/v2/coins/algo.ts:353]: Add js doc

File: modules/core/src/v2/coins/algo.ts [hunk 3]
```diff
@@ -348,6 +350,23 @@ export class Algo extends BaseCoin {
...
+    return SeedValidator.isValidEd25519SeedForCoin(seed, CoinFamily.XLM);
+  }
+
+  convertFromStellarSeed(seed: string): string | null {
```
Comment (rliberal-simtlix) [inline modules/core/src/v2/coins/algo.ts:357]: Add js doc

**Review by rliberal-simtlix — COMMENTED** [review]
Changes looks good just nits

File: modules/account-lib/src/coin/algo/seedEncoding.ts (hunk 1 above)
Comment (CapnMigraine) [reply modules/account-lib/src/coin/algo/seedEncoding.ts:51]: done

File: modules/core/src/v2/coins/algo.ts (hunk 2 above)
Comment (CapnMigraine) [reply modules/core/src/v2/coins/algo.ts:359]: done

File: modules/core/src/v2/coins/algo.ts (hunk 3 above)
Comment (CapnMigraine) [reply modules/core/src/v2/coins/algo.ts:369]: done

**Review by rliberal-simtlix — APPROVED** [review]

**Review by JulioAchaval-SimTLiX — APPROVED** [review]

**Review by smaria95 — APPROVED** [review]
LGTM

**Review by Krin2 — APPROVED** [review]

File: modules/account-lib/src/coin/algo/seedEncoding.ts [hunk 4]
```diff
@@ -0,0 +1,88 @@
+import * as base32 from 'hi-base32';
+import sha512 from 'js-sha512';
+import { Seed } from './ifaces';
+
+const SEED_BYTES_LENGTH = 32;
+const ALGORAND_SEED_BYTE_LENGTH = 36;
+const ALGORAND_CHECKSUM_BYTE_LENGTH = 4;
+const ALGORAND_SEED_LENGTH = 58;
+
+const MALFORMED_SEED_ERROR = new Error('seed seems to be malformed');
```
Comment (tylerlevine) [inline modules/account-lib/src/coin/algo/seedEncoding.ts:10]: nit: create the error where it gets thrown, otherwise the stack trace won't be very helpful.

**Review by tylerlevine — CHANGES_REQUESTED** [review]

**Review by tylerlevine — CHANGES_REQUESTED** [review]
please rebase due to breaking changes in underlying utxolib library. I don't think this PR should be affected by this but I think it would be good to rebase anyway just in case.

**Pull request closed without merging**